api/s2n.h

```diff
@@ -122,12 +123,49 @@ extern int s2n_connection_wipe(struct s2n_connection *conn);
 extern int s2n_connection_free(struct s2n_connection *conn);
 extern int s2n_shutdown(struct s2n_connection *conn, s2n_blocked_status *blocked);
 
+typedef enum { S2N_CERT_AUTH_NONE, S2N_CERT_AUTH_REQUIRED } s2n_cert_auth_type;
+extern int s2n_config_get_client_auth_type(struct s2n_config *config, s2n_cert_auth_type *client_auth_type);
+extern int s2n_config_set_client_auth_type(struct s2n_config *config, s2n_cert_auth_type client_auth_type);
+extern int s2n_connection_get_client_auth_type(struct s2n_connection *conn, s2n_cert_auth_type *client_auth_type);
+extern int s2n_connection_set_client_auth_type(struct s2n_connection *conn, s2n_cert_auth_type client_auth_type);
+extern int s2n_connection_get_client_cert_chain(struct s2n_connection *conn, uint8_t **der_cert_chain_out, uint32_t *cert_chain_len);
+
```

Comment:
> If you were running s2n in client mode, you would be setting the auth type for the server. I think of it as setting the "peer" auth type, where "peer" would mean different sides of the connection depending on which mode you were in.

Do you mean setting CERT_AUTH_NONE on the client connection would skip certificate validation of the Server's Cert? I don't think we want to do that. If we were to go with this API it would likely only allow setting CERT_AUTH_NONE on the server connection, return an error on the client connection, and also require peeking at the next TLS Record Type in order to have the Client dynamically handle Cert Requests from the server (this is only a couple of lines of code, but would be the one and only place where we are dynamically changing the TLS State Machine haver the Client and Server Hello's, and may make the Formal Verification of the State Machine more difficult).